README.md

```diff
@@ -15,17 +15,28 @@ Implement the tool `st` that does exact pattern matching using a suffix tree. Te
 
 ## Specify if you have used a linear time or quadratic time algorithm.
 
+I have implemented McCreight's O(n) algorithm.
+
 ## Insights you may have had while implementing and comparing the algorithms.
 
+There is a lot to keep track of with the various nodes and suffix links. 
+
 ## Problems encountered if any.
 
+I used a lot of time to understand the algorithm. 
+After, it was difficult to implement in python because some things in C are so different.
```

Comment:
I would recommend that you try to understand the algorithm separate from reading code. Code is only interesting if it is programming tricks and techniques you are interested in; algorithms are best understood by drawing various cases and working them out on paper.